src/analysis/AnalysisClient.js

```diff
@@ -93,21 +95,71 @@ class AnalysisClient {
     analyzed_items_file.end();
   }
 
+
+  assignRepresentativeWeightToItem(pairedProductsList){
+    /*
+    If we have seen at least five item in a category, assign 3/4 of the max weight in a category to an item 
+    in that category whose weight is unknown
+    The choice of 3/4 of max is arbitrary and can be changed in the future.    
+    If we have seen less than 5, use the max weight instead because we don't know enough about weights in category
+    */
+    let representativeCount = 5;
+    let representativeWeight = 0.75;
+    let categoriesRepWeight  = {};
+    let categoriesCount  = {};
+    
+    pairedProductsList.products.forEach(function(pairedProduct){
+      //if weight is known and categoryId is defined
+      if (pairedProduct.amazonProd.dimensions.weight != 'UNKNOWN' && pairedProduct.amazonProd.bestSalesRanking.categoryId) {
+        if (categoriesRepWeight.hasOwnProperty(pairedProduct.amazonProd.bestSalesRanking.categoryId)) {
+          if (categoriesRepWeight[pairedProduct.amazonProd.bestSalesRanking.categoryId] < Math.ceil(parseFloat(pairedProduct.amazonProd.dimensions.weight['C$']))) {
+            categoriesRepWeight[pairedProduct.amazonProd.bestSalesRanking.categoryId] = Math.ceil(parseFloat(pairedProduct.amazonProd.dimensions.weight['C$']));
+          }
+          categoriesCount[pairedProduct.amazonProd.bestSalesRanking.categoryId] +=1;
+        }
+        else {
+          categoriesRepWeight[pairedProduct.amazonProd.bestSalesRanking.categoryId] = Math.ceil(parseFloat(pairedProduct.amazonProd.dimensions.weight['C$']));
+          categoriesCount[pairedProduct.amazonProd.bestSalesRanking.categoryId] =1;
+        }
+      }
+    });
+
+    for (var category in categoriesRepWeight) {
+      if (categoriesCount[category] >= representativeCount) {
+        categoriesRepWeight[category] = Math.ceil(categoriesRepWeight[category]*representativeWeight);
+      }
+    }
+    return categoriesRepWeight;
+  }
+
   /* does simple analysis of cost per item based on shipping and weight */
   getSimpleCostAnalysis(pairedProductsList) {
     let that = this;
+    //use representative weight of each category to assign weight values to items with unknown weights
+    //this helps to compute their shipping price and see what is the profit potential for such items.
+    let representativeWeights = that.assignRepresentativeWeightToItem(pairedProductsList);
     let analyzedProduct;
-
+       
     pairedProductsList.products.forEach(function(pairedProduct){
-      // We already know all amazon products have a known price and weight since they are filtered 
-      // in the AmazonProduct class.
-      analyzedProduct = that.getAnalyzedProductInfo(pairedProduct);
-      // Only interested in items with a high margin of profit.
-      if (analyzedProduct.basePercentROI >= 25) {
-        that.analyzedProductsInfo.push(analyzedProduct);
+      //checks if amazon price is known
+      if (pairedProduct.amazonProd.price != 'UNKNOWN') {
```

Comment:
At this point, we have already filtered out all amazon products with an unknown price. That was done here - https://github.com/sulaimonlasisi/clearance-app/blob/master/src/amazon/AmazonProductList.js#L44